### ISSUE TITLE
[Feature Request]: Check-in when creating asset

### ISSUE THREAD

**Issue by NorskNoobing** [grokability/snipe-it#12893]
### Is your feature request related to a problem? Please describe.

I'd like the option to check in assets to a location in the asset creator instead of just checking the assets out. This is useful when I'm creating assets that's currently in our storage.

In order to fix this today, I'll have to temporarily check the asset out to an user or location, then press the checkin button and assign it to our storage.

### Describe the solution you'd like

The perfect solution would be to just add a new option when creating an asset, where you can select if you want to check-in, and then select a location, or check-out which would work just as it does today.

### Describe alternatives you've considered

We've tried using default location on the assets, but this won't work as intended if the user doesn't have any location assigned to themselves. So when you check the asset out to the user, it'll still show the default location in the asset-list, because the user doesn't have any location:
![bilde](https://user-images.githubusercontent.com/39239702/233914713-b41e2323-2ca6-4546-9efe-41284f50ad5a.png)

**Comment by snipe** [comment]
You could check it out to a location on create and then check it in and back out to a user when it's ready to be deployed.

**Comment by NorskNoobing** [comment]
Yeah, that's the routine I've had for myself. The problem is that we have around 50-60 agents that should use Snipe-IT, and I know that someone's gonna mess up with a convoluted routine like this.

We also probably have around 3800 assets on each location, and there are 23 or so locations.

So unfortunately I think that this isn't really going to work 😕
